repository/gorm/models.go

```diff
@@ -0,0 +1,89 @@
+package gorm
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	open_asset_model "github.com/owasp-amass/asset-db/open_asset_model"
+	"github.com/owasp-amass/asset-db/types"
+
+	"gorm.io/datatypes"
+)
+
+type Asset struct {
+	ID        int64     `gorm:"primaryKey;autoIncrement:true"`
+	CreatedAt time.Time `gorm:"type:datetime"`
+	Type      string
+	Content   datatypes.JSON
+}
+
+func (a Asset) Parse() (types.Asset, error) {
```

Comment:
where would this be used?

Reply:
It is used when searching for an asset.

We define different types to manage them at the database level and at the application level. So when we get an asset from the database we need to rebuild again the type with the stored JSON, and for that, we create this function (I think the concept is type assertion)

Reply:
Asset type for the application level that lives in open-asset-model (only one discovery without ID)
```
Asset type interface {
	AssetType() Asset type
	JSON() ([]byte, error)
}
```
StoredAsset is an Asset + ID that lives in asset-db and is the structure that returns the asset repositories functions.
```
type StoredAsset struct {
	ID string
	Asset Asset
}
```
And Asset struct that lives in asset-db and is only handled by the repositories to store the entity.
```
type Asset struct {
	ID        int64     `gorm:"primaryKey;autoIncrement:true"`
	CreatedAt time.Time `gorm:"type:datetime"`
	Type      string
	Content   datatypes.JSON
}
```